server/src/test/java/org/opensearch/extensions/rest/RestInitializeExtensionActionTests.java

```diff
@@ -121,4 +130,90 @@ public void testRestInitializeExtensionActionFailure() throws Exception {
         );
     }
 
+    public void testRestInitializeExtensionActionResponseWithAdditionalSettings() throws Exception {
+        Setting boolSetting = Setting.boolSetting("boolSetting", false, Setting.Property.ExtensionScope);
+        Setting stringSetting = Setting.simpleString("stringSetting", "default", Setting.Property.ExtensionScope);
+        Setting intSetting = Setting.intSetting("intSetting", 0, Setting.Property.ExtensionScope);
+        Setting listSetting = Setting.listSetting(
+            "listSetting",
+            List.of("first", "second", "third"),
+            Function.identity(),
+            Setting.Property.ExtensionScope
+        );
+        ExtensionsManager extensionsManager = new ExtensionsManager(Set.of(boolSetting, stringSetting, intSetting, listSetting));
+        ExtensionsManager spy = spy(extensionsManager);
+
+        // optionally, you can stub out some methods:
+        when(spy.getAdditionalSettings()).thenCallRealMethod();
+        Mockito.doCallRealMethod().when(spy).loadExtension(any(ExtensionsSettings.Extension.class));
+        Mockito.doNothing().when(spy).initialize();
+        RestInitializeExtensionAction restInitializeExtensionAction = new RestInitializeExtensionAction(spy);
+        final String content = "{\"name\":\"ad-extension\",\"uniqueId\":\"ad-extension\",\"hostAddress\":\"127.0.0.1\","
+            + "\"port\":\"4532\",\"version\":\"1.0\",\"opensearchVersion\":\"3.0.0\","
```

Comment:
Can you pull the OS version from `Version.CURRENT.toString()`?

Reply:
Changed to `Version.CURRENT.toString()`